tests/tornado/test_connection.py

```diff
@@ -128,3 +129,52 @@ def build_raw_message(self, context, args, is_completed=True):
 
     assert client_peer.total_outbound_pendings == 0
     assert server_peer.total_outbound_pendings == 0
+
+
+@pytest.mark.gen_test
+def test_client_connection_change_callback():
+    server = TChannel('server')
+    server.listen()
+
+    @server.raw.register
+    def hello(request):
+        return 'hi'
+
+    client = TChannel('client')
+    count = [0]
+
+    def test_cb(peer):
+        count[0] += 1
+
+    client._dep_tchannel.peers.get(
+        server.hostport)._on_conn_change_cb = test_cb
```

Comment:
No. Test these directly on the Peer or PeerGroup object. Private attribute access in a test is generally an indicator that you need to change the test.